### DISCUSSION TITLE
[Feature]: Displaying Measured Distances on Aligning Guidelines

### DISCUSSION THREAD

**Discussion by hasanmumin** [fabricjs/fabric.js#10347] [General]
### CheckList

- [x] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have searched and referenced existing issues, feature requests and discussions
- [X] I am filing a **FEATURE** request.

### Description

Currently, Fabric.js allows for drawing aligning guidelines (lines) on the canvas to help with positioning and aligning objects. Enhancing this feature to also show the distance between two points directly on the canvas would provide immediate, visual feedback. Developers and designers could quickly see the exact measured distances, streamlining the design and layout process.

### Current State

1. Fabric.js provides functionality to draw lines and overlays for guiding alignment.
2. There is no built-in method to display numeric distances (e.g., “20px”) along these guidelines.
3. Users must rely on external calculations or debug logs to determine distances between points.

### Additional Context

By implementing a distance-measurement feature:

1. Users would no longer need to switch contexts or rely solely on external measurements.
2. The feature would be particularly useful in design applications, or any scenario where precise spacing and measurement are important.
3. Offering configurable options (font size, color, positioning, and units) would give developers control over the display and improve the overall utility of the guideline feature.

**Comment by asturur** [comment]
That is an app feature that can be built by the developer extending the current guidelines.
I do not think fabricJS should support that by default, is outside the scope of what it does in my opinion.

**Reply by hasanmumin** [reply]
I thought Aligning Guidelines is an official extension, thats why I asked. 
https://github.com/fabricjs/fabric.js/tree/master/extensions/aligning_guidelines
